collection/rancher/v2.x/logs-collector/rancher2_logs_collector.sh

```diff
@@ -1,311 +1,528 @@
+    done
+    if [ -d /etc/kubernetes/.tmp ]; then
+      mkdir -p $TMPDIR/k8s/tmpcerts
+      TMPCERTS=$(find /etc/kubernetes/.tmp -type f -name *.pem | grep -v "\-key\.pem$")
+      for TMPCERT in $TMPCERTS; do
+        openssl x509 -in $TMPCERT -text -noout > $TMPDIR/k8s/tmpcerts/$(basename $TMPCERT) 2>&1
+      done
+    fi
   fi
-fi
 
-# etcd
-echo "Collecting etcd info"
-mkdir -p $TMPDIR/etcd
-# /var/lib/etcd contents
-if [ -d /var/lib/etcd ]; then
-  find /var/lib/etcd -type f -exec ls -la {} \; > $TMPDIR/etcd/findvarlibetcd 2>&1
-elif [ -d /opt/rke/var/lib/etcd ]; then
-  find /opt/rke/var/lib/etcd -type f -exec ls -la {} \; > $TMPDIR/etcd/findoptrkevarlibetcd 2>&1
-fi
+}
 
-# /opt/rke contents
-if [ -d /opt/rke/etcd-snapshots ]; then
-  find /opt/rke/etcd-snapshots -type f -exec ls -la {} \; > $TMPDIR/etcd/findoptrkeetcdsnaphots 2>&1
-fi
+etcd() {
+
+  techo "Collecting etcd info"
+  mkdir -p $TMPDIR/etcd
+  if [ -d /var/lib/etcd ]; then
+    find /var/lib/etcd -type f -exec ls -la {} \; > $TMPDIR/etcd/findvarlibetcd 2>&1
+  elif [ -d /opt/rke/var/lib/etcd ]; then
+    find /opt/rke/var/lib/etcd -type f -exec ls -la {} \; > $TMPDIR/etcd/findoptrkevarlibetcd 2>&1
+  fi
 
-# etcd
-if docker ps --format='{{.Names}}' | grep -q ^etcd$ >/dev/null 2>&1; then
-  echo "Collecting etcdctl output"
-  PARAM=""
-  # Check for older versions with incorrectly set ETCDCTL_ENDPOINT vs the correct ETCDCTL_ENDPOINTS
-  # If ETCDCTL_ENDPOINTS is empty, its an older version
-  if [ -z $(docker exec etcd printenv ETCDCTL_ENDPOINTS) ]; then
-    ENDPOINT=$(docker exec etcd printenv ETCDCTL_ENDPOINT)
-    if echo $ENDPOINT | grep -vq 0.0.0.0; then
-      PARAM="--endpoints=$(docker exec etcd printenv ETCDCTL_ENDPOINT)"
+  # /opt/rke contents
+  if [ -d /opt/rke/etcd-snapshots ]; then
+    find /opt/rke/etcd-snapshots -type f -exec ls -la {} \; > $TMPDIR/etcd/findoptrkeetcdsnaphots 2>&1
+  fi
+
+  if docker ps --format='{{.Names}}' | grep -q ^etcd$ >/dev/null 2>&1; then
+    techo "Collecting etcdctl output"
+    PARAM=""
+    # Check for older versions with incorrectly set ETCDCTL_ENDPOINT vs the correct ETCDCTL_ENDPOINTS
+    # If ETCDCTL_ENDPOINTS is empty, its an older version
+    if [ -z $(docker exec etcd printenv ETCDCTL_ENDPOINTS) ]; then
+      ENDPOINT=$(docker exec etcd printenv ETCDCTL_ENDPOINT)
+      if echo $ENDPOINT | grep -vq 0.0.0.0; then
+        PARAM="--endpoints=$(docker exec etcd printenv ETCDCTL_ENDPOINT)"
+      fi
     fi
+    docker exec etcd sh -c "etcdctl $PARAM member list"  > $TMPDIR/etcd/memberlist 2>&1
+    docker exec etcd etcdctl endpoint status --endpoints=$(docker exec etcd /bin/sh -c "etcdctl $PARAM member list | cut -d, -f5 | sed -e 's/ //g' | paste -sd ','") --write-out table > $TMPDIR/etcd/endpointstatus 2>&1
+    docker exec etcd etcdctl endpoint health --endpoints=$(docker exec etcd /bin/sh -c "etcdctl $PARAM member list | cut -d, -f5 | sed -e 's/ //g' | paste -sd ','") > $TMPDIR/etcd/endpointhealth 2>&1
```

Comment:
This give a critical error:
```
ip-10-0-99-10:/tmp # cat etcd/endpointhealth
2020-05-10 19:42:05.730393 C | pkg/flags: conflicting environment variable "ETCDCTL_ENDPOINTS" is shadowed by corresponding command-line flag (either unset environment variable or disable flag)
```

Reply:
Tracking this in https://github.com/rancherlabs/support-tools/issues/31